consensus/wal.go

```diff
@@ -109,13 +116,35 @@ func (wal *baseWAL) OnStart() error {
 		wal.WriteSync(EndHeightMessage{0})
 	}
 	err = wal.group.Start()
+	wal.syncTicker = time.NewTicker(walSyncInterval)
+	go wal.processSyncTicks()
 	return err
 }
 
+// processSyncTicks allows us to periodically attempt to sync the WAL to disk.
+func (wal *baseWAL) processSyncTicks() {
+	for {
+		select {
+		case <-wal.syncTicker.C:
+			if err := wal.group.Flush(); err != nil {
+				wal.Logger.Error("Failed to flush WAL", "err", err)
+				wal.testChan <- "ERROR"
+			} else {
+				wal.Logger.Debug("Successfully sync'd WAL to disk")
```

Comment:
not sure we need this

Reply:
I need some way of checking whether or not the sync happened successfully in the test, otherwise I'm just checking if the ticker actually ticked.

Reply:
> not sure we need this

Oh, do you mean the `wal.Logger.Debug()` call?

Reply:
yes